### ISSUE TITLE
Modded WSA launch fail ANDROID 13

### ISSUE THREAD

**Issue by kevstar0** [LSPosed/MagiskOnWSALocal#586]
### Steps to reproduce/复现步骤

1. use MagiskOnWsaLocal using "run.sh"
2. choose stable versions, root/ don't root WSA include Magisk, GAPPS & exclude Amazon App store don't compress the output
3. Install WSA using run.bat file from the output folder
4.Wait for WSA to launch

### Expected behaviour/预期行为

The build will do it's work just fine and produce the modded WSA from the output folder

### Actual behaviour/实际行为

Upon installing the modded WSA it refuses to launch instead it exits silently.
a few notifications will pop out indicating WSA system upgrades and to configure your keyboard but that's just that.

### MagiskOnWSALocal commit full SHA/MagiskOnWSALocal 提交的完整哈希

aefda92

### Linux distribution info/Linux 发行版信息

Ubuntu

### Windows version/Windows 版本

22H2

### Build Parameters/构建参数

sir@CyberLord:/mnt/c/Android/MagiskOnWSALocal/output$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 22.04.2 LTS
Release:        22.04
Codename:       jammy

### Version requirement/版本要求

- [X] I am using latest commit of MagiskOnWSALocal/我正在使用最新 commit 的版本

### Logs/日志

<details>

[//]: <> (Don't modify above)
It loads & quits
```
It loads & quits
![image](https://github.com/LSPosed/MagiskOnWSALocal/assets/38152088/2d513333-42d0-4742-a2c3-2e10b8adf9cb)

```
[//]: <> (Don't modify below)
</details>

**Comment by YT-Advanced** [comment]
@kevstar0 Can you export the log in this dir please:
`%LOCALAPPDATA%\Packages\MicrosoftCorporationII.WindowsSubsystemForAndroid_8wekyb3d8bbwe\LocalState\diagnostics\logcat`

**Comment by kevstar0** [comment]
> @kevstar0 Can you export the log in this dir please: `%LOCALAPPDATA%\Packages\MicrosoftCorporationII.WindowsSubsystemForAndroid_8wekyb3d8bbwe\LocalState\diagnostics\logcat`

How do I navigate to such directory? it doesn't exist anywhere on my machine

**Comment by YT-Advanced** [comment]
@kevstar0 If you still have WSA installed, open Win+R, paste it and then Enter

**Comment by kevstar0** [comment]
> @kevstar0 If you still have WSA installed, open Win+R, paste it and then Enter

here is contents of the log file

--------- beginning of main
06-29 16:27:33.938    22    22 I lowmemorykiller: Using psi monitors for memory pressure detection
06-29 16:27:33.939    22    22 I lowmemorykiller: Process polling is supported
06-29 16:27:33.939    22    22 I lowmemorykiller: Process reaper initialized with 2 threads in the pool
--------- beginning of events
06-29 16:27:34.102    23    23 I snet_event_log: [121035042,-1,]
06-29 16:27:34.581    23    23 W SELinux : Multiple same specifications for android.hardware.camera.provider.ICameraProvider/internal/0.
06-29 16:27:34.581    23    23 I auditd  : Multiple same specifications for android.hardware.camera.provider.ICameraProvider/internal/0.
06-29 16:27:34.581    23    23 I SELinux : SELinux: Loaded service_contexts from:
06-29 16:27:34.581    23    23 I auditd  : SELinux: Loaded service_contexts from:
06-29 16:27:34.581    23    23 I SELinux :     /system/etc/selinux/plat_service_contexts
06-29 16:27:34.581    23    23 I auditd  :     /system/etc/selinux/plat_service_contexts
06-29 16:27:34.581    23    23 I SELinux :     /system_ext/etc/selinux/system_ext_service_contexts
06-29 16:27:34.581    23    23 I auditd  :     /system_ext/etc/selinux/system_ext_service_contexts
06-29 16:27:34.581    23    23 I SELinux :     /product/etc/selinux/product_service_contexts
06-29 16:27:34.581    23    23 I auditd  :     /product/etc/selinux/product_service_contexts
06-29 16:27:34.581    23    23 I SELinux :     /vendor/etc/selinux/vendor_service_contexts
06-29 16:27:34.581    23    23 I auditd  :     /vendor/etc/selinux/vendor_service_contexts
06-29 16:27:34.679    24    24 I hwservicemanager: getFrameworkHalManifest: Reading VINTF information.
06-29 16:26:46.010    21    21 I auditd  : type=2000 audit(0.0:1): state=initialized audit_enabled=0 res=1
06-29 16:26:46.010    21    21 W auditd  : type=2000 audit(0.0:1): state=initialized audit_enabled=0 res=1
06-29 16:27:09.360    21    21 I auditd  : type=1403 audit(0.0:2): auid=4294967295 ses=4294967295 lsm=selinux res=1
06-29 16:27:09.360    21    21 W auditd  : type=1403 audit(0.0:2): auid=4294967295 ses=4294967295 lsm=selinux res=1
06-29 16:27:09.490    21    21 I auditd  : type=1404 audit(0.0:3): enforcing=1 old_enforcing=0 auid=4294967295 ses=4294967295 enabled=1 old-enabled=1 lsm=selinux res=1
06-29 16:27:09.490    21    21 W auditd  : type=1404 audit(0.0:3): enforcing=1 old_enforcing=0 auid=4294967295 ses=4294967295 enabled=1 old-enabled=1 lsm=selinux res=1
06-29 16:27:27.490    79    79 I auditd  : type=1400 audit(0.0:4): avc: denied { sys_ptrace } for comm="init" capability=19 scontext=u:r:init:s0 tcontext=u:r:init:s0 tclass=capability permissive=0
06-29 16:27:27.490    79    79 W init    : type=1400 audit(0.0:4): avc: denied { sys_ptrace } for capability=19 scontext=u:r:init:s0 tcontext=u:r:init:s0 tclass=capability permissive=0
06-29 16:27:27.490    79    79 I auditd  : type=1400 audit(0.0:5): avc: denied { sys_ptrace } for comm="init" capability=19 scontext=u:r:init:s0 tcontext=u:r:init:s0 tclass=capability permissive=0
06-29 16:27:27.490    79    79 W init    : type=1400 audit(0.0:5): avc: denied { sys_ptrace } for capability=19 scontext=u:r:init:s0 tcontext=u:r:init:s0 tclass=capability permissive=0
06-29 16:27:27.490    79    79 I auditd  : type=1400 audit(0.0:6): avc: denied { sys_ptrace } for comm="init" capability=19 scontext=u:r:init:s0 tcontext=u:r:init:s0 tclass=capability permissive=0
06-29 16:27:27.490    79    79 W init    : type=1400 audit(0.0:6): avc: denied { sys_ptrace } for capability=19 scontext=u:r:init:s0 tcontext=u:r:init:s0 tclass=capability permissive=0
06-29 16:27:27.490    79    79 I auditd  : type=1400 audit(0.0:7): avc: denied { sys_ptrace } for comm="init" capability=19 scontext=u:r:init:s0 tcontext=u:r:init:s0 tclass=capability permissive=0
06-29 16:27:27.490    79    79 W init    : type=1400 audit(0.0:7): avc: denied { sys_ptrace } for capability=19 scontext=u:r:init:s0 tcontext=u:r:init:s0 tclass=capability permissive=0
06-29 16:27:27.490    79    79 I auditd  : type=1400 audit(0.0:8): avc: denied { sys_ptrace } for comm="init" capability=19 scontext=u:r:init:s0 tcontext=u:r:init:s0 tclass=capability permissive=0
06-29 16:27:27.490    79    79 W init    : type=1400 audit(0.0:8): avc: denied { sys_ptrace } for capability=19 scontext=u:r:init:s0 tcontext=u:r:init:s0 tclass=capability permissive=0
06-29 16:27:34.781    24    24 I hwservicemanager: getDeviceHalManifest: Reading VINTF information.
06-29 16:27:34.800    24    24 I hwservicemanager: getDeviceHalManifest: Successfully processed VINTF information
06-29 16:27:34.800    24    24 I hwservicemanager: getFrameworkHalManifest: Successfully processed VINTF information
06-29 16:27:34.801    24    24 I hwservicemanager: hwservicemanager is ready now.
--------- beginning of system
06-29 16:27:34.918    25    25 I vold    : Vold 3.0 (the awakening) firing up
06-29 16:27:34.973    25    25 I Checkpoint: Validating checkpoint on /dev/block/sde
06-29 16:27:34.994    25    25 E Checkpoint: No magic
06-29 16:27:35.004    24    24 E cutils-trace: Error opening trace file: No such file or directory (2)
06-29 16:27:35.004    24    24 I hwservicemanager: getTransport: Cannot find entry android.hardware.boot@1.0::IBootControl/default in either framework or device VINTF manifest.
06-29 16:27:39.932    39    39 I virtio_ports_create_links: Completed successfully.
06-29 16:27:40.255    23    23 I servicemanager: getDeviceHalManifest: Reading VINTF information.
06-29 16:27:40.256    23    23 I servicemanager: getDeviceHalManifest: Successfully processed VINTF information
06-29 16:27:40.256    23    23 I servicemanager: getFrameworkHalManifest: Reading VINTF information.
06-29 16:27:40.257    23    23 I servicemanager: getFrameworkHalManifest: Successfully processed VINTF information
06-29 16:27:40.257    23    23 I servicemanager: Found android.system.suspend.ISystemSuspend/default in framework VINTF manifest.
06-29 16:27:40.257    24    24 I hwservicemanager: getTransport: Cannot find entry android.system.suspend@1.0::ISystemSuspend/default in either framework or device VINTF manifest.
06-29 16:27:40.257    41    41 E HidlServiceManagement: Service android.system.suspend@1.0::ISystemSuspend/default must be in VINTF manifest in order to register/get.
06-29 16:27:40.257    41    41 I android.system.suspend@1.0-service: system-suspend HIDL hal not supported, use the AIDL suspend hal for requesting wakelocks
06-29 16:27:40.330    49    49 I fsverity_init: LoadKeyFromFile path=/system/etc/security/fsverity/fsverity-release.x509.der keyname=fsv_system_0
06-29 16:27:40.387    49    49 I fsverity_init: LoadKeyFromFile path=/product/etc/security/fsverity/gms_fsverity_cert.der keyname=fsv_product_0
06-29 16:27:40.663    51    51 I provision_apex: Provisioning apex
06-29 16:27:40.671    51    51 I provision_apex: Not opening /mnt/shared_tmp/apex (No such file or directory)
06-29 16:27:40.716    25    25 I Checkpoint: cp_prepareCheckpoint called
06-29 16:27:40.757    25    25 I vold    : fscrypt_initialize_systemwide_keys
06-29 16:27:40.837    48    48 I RemoteConnectionManager: RemoteConnectionManager service is starting for instance default
06-29 16:27:40.837    45    45 I GSKServerConnection: GSKServerConnection service is starting
06-29 16:27:40.838    23    23 I servicemanager: Found windows.services.graphics.IGSKServerConnection/default in device VINTF manifest.
06-29 16:27:40.838    45    45 I GSKServerConnection: Service default is ready
06-29 16:27:40.840    23    23 I servicemanager: Found windows.services.rcm.IRemoteConnectionManager/default in device VINTF manifest.
06-29 16:27:40.840    48    48 I RemoteConnectionManager: Service default is ready
06-29 16:27:40.870    23    23 I servicemanager: Since 'android.system.keystore2.IKeystoreService/default' could not be found, trying to start it as a lazy AIDL service
06-29 16:27:40.872    23    55 W libc    : Unable to set property "ctl.interface_start" to "aidl/android.system.keystore2.IKeystoreService/default": error code: 0x20
06-29 16:27:40.872    23    55 I servicemanager: Tried to start aidl service android.system.keystore2.IKeystoreService/default as a lazy service, but was unable to. Usually this happens when a service is not installed, but if the service is intended to be used as a lazy service, then it may be configured incorrectly.
06-29 16:27:41.097    43    43 I keymaster:  patchlevel string does not match expected format.  Using patchlevel 0
06-29 16:27:41.098    43    43 I HidlServiceManagement: Registered android.hardware.keymaster@4.1::IKeymasterDevice/default
06-29 16:27:41.098    43    43 I HidlServiceManagement: Removing namespace from process name android.hardware.keymaster@4.1-service to keymaster@4.1-service.
06-29 16:27:41.870    25    25 W ServiceManager: Waited one second for android.system.keystore2.IKeystoreService/default (is service started? are binder threads started and available?)
06-29 16:27:42.033    23    23 I servicemanager: Since 'android.system.keystore2.IKeystoreService/default' could not be found, trying to start it as a lazy AIDL service
06-29 16:27:42.035    23    56 W libc    : Unable to set property "ctl.interface_start" to "aidl/android.system.keystore2.IKeystoreService/default": error code: 0x20
06-29 16:27:42.035    23    56 I servicemanager: Tried to start aidl service android.system.keystore2.IKeystoreService/default as a lazy service, but was unable to. Usually this happens when a service is not installed, but if the service is intended to be used as a lazy service, then it may be configured incorrectly.
06-29 16:27:42.321    42    42 I keystore2: keystore2_main: Keystore2 is starting.
06-29 16:27:42.321    42    42 I keystore2: keystore2_main: Setting up sqlite logging for keystore2
06-29 16:27:42.477    42    42 W keystore2: keystore2::metrics_store: In update_keystore_crash_sysprop: Failed to read the existing system property due to: In read_keystore_crash_count: Property not set..Therefore, keystore crashes will not be logged.
06-29 16:27:42.561    42    42 I keystore2: keystore2_main: Starting thread pool now.
06-29 16:27:42.561    42    42 I keystore2: getDeviceHalManifest: Reading VINTF information.
06-29 16:27:42.562    42    42 I keystore2: getDeviceHalManifest: Successfully processed VINTF information
06-29 16:27:42.564    42    42 I keystore2: found android.hardware.keymaster@4.0::IKeymasterDevice with interface name default and seclevel SOFTWARE
06-29 16:27:42.565    24    24 I hwservicemanager: getTransport: Cannot find entry android.hardware.keymaster@3.0::IKeymasterDevice/default in either framework or device VINTF manifest.
06-29 16:27:42.565    42    42 W keystore2: No secure Keymaster implementation found, but device offers insecure Keymaster HAL. Using as default.
06-29 16:27:42.565    42    42 I keymaster:  patchlevel string does not match expected format.  Using patchlevel 0
06-29 16:27:42.565    42    42 I keystore2: keystore2::globals: Add emulation wrapper around Keymaster device for security level: SecurityLevel(1)
06-29 16:27:42.565    42    42 I keymaster:  patchlevel string does not match expected format.  Using patchlevel 0
06-29 16:27:42.566    23    23 I servicemanager: Found android.system.keystore2.IKeystoreService/default in framework VINTF manifest.
06-29 16:27:42.566    23    23 I BpBinder: onLastStrongRef automatically unlinking death recipients: <uncached descriptor>
06-29 16:27:42.590    42    42 I keystore2: keystore2_main: Not publishing IRemotelyProvisionedKeyPool service: In new_native_binder: Failed to get TEE Remote Provisioner instance.
06-29 16:27:42.590    42    42 I keystore2: 
06-29 16:27:42.590    42    42 I keystore2: Caused by:
06-29 16:27:42.590    42    42 I keystore2:     0: In get_remotely_provisioned_component.
06-29 16:27:42.590    42    42 I keystore2:     1: In connect_remotely_provisioned_component.
06-29 16:27:42.590    42    42 I keystore2:     2: Error::Km(ErrorCode(-68))
06-29 16:27:42.590    42    42 I keystore2: keystore2_main: Successfully registered Keystore 2.0 service.
06-29 16:27:42.590    42    42 I keystore2: keystore2_main: Joining thread pool now.
06-29 16:27:42.590    42    57 I keystore2: keystore2::shared_secret_negotiation: Shared secret negotiation concluded successfully.
06-29 16:27:42.590    42    57 I keystore2: keystore2::shared_secret_negotiation: No StrongBox Keymint available, so no RoT transfer
06-29 16:27:42.652    42    42 I SELinux : SELinux: Loaded keystore2_key_contexts from:
06-29 16:27:42.652    42    42 I auditd  : SELinux: Loaded keystore2_key_contexts from:
06-29 16:27:42.652    42    42 I SELinux :     /system/etc/selinux/plat_keystore2_key_contexts
06-29 16:27:42.652    42    42 I auditd  :     /system/etc/selinux/plat_keystore2_key_contexts
06-29 16:27:43.545    25    25 I vold    : Wrote system DE key reference to:/data/unencrypted/ref
06-29 16:27:44.070    25    25 I vold    : Wrote per boot key reference to:/data/unencrypted/per_boot_ref
06-29 16:27:46.167    25    25 I incfs   : Initial API level of the device: 30
06-29 16:27:46.167    25    25 I incfs   : ro.incremental.enable: true
06-29 16:27:46.167    25    25 I incfs   : isFsAvailable: true
06-29 16:27:46.167    25    25 I incfs   : isFsAvailable: true
06-29 16:27:46.167    25    25 I incfs   : IncFs_Features: v2
06-29 16:27:46.168    42    42 I keystore2: keystore2::maintenance: In early_boot_ended.
06-29 16:27:46.456    62    62 I tombstoned: tombstoned successfully initialized
06-29 16:27:46.515    42    42 I keystore2: keystore2::globals: Touching Keystore 2.0 database for this first time since boot.
06-29 16:27:46.515    42    42 I keystore2: keystore2::globals: Calling cleanup leftovers.
06-29 16:27:46.516    42    42 I keystore2: keystore2::super_key: Starting boot level watcher.
06-29 16:27:46.517    42    65 I keystore2: keystore2::super_key: Read keystore.boot_level value 30
06-29 16:27:46.517    42    42 I keystore2: keystore2::maintenance: Call to earlyBootEnded succeeded for security level TRUSTED_ENVIRONMENT.
06-29 16:27:46.517    42    42 E keystore2: keystore2::maintenance: Call to earlyBootEnded failed for security level STRONGBOX: In call_with_watchdog: getting keymint device.
06-29 16:27:46.517    42    42 E keystore2: keystore2::error: In call_with_watchdog: getting keymint device
06-29 16:27:46.517    42    42 E keystore2: 
06-29 16:27:46.517    42    42 E keystore2: Caused by:
06-29 16:27:46.517    42    42 E keystore2:     0: In get_keymint_device.
06-29 16:27:46.517    42    42 E keystore2:     1: In connect_keymint: Trying to get Legacy wrapper.
06-29 16:27:46.517    42    42 E keystore2:     2: Error::Km(ErrorCode(-68))
06-29 16:27:46.517    42    65 I keystore2: keystore2::super_key: Read keystore.boot_level value 30
06-29 16:27:46.517    25    25 E vold    : keystore2 Keystore earlyBootEnded returned service specific error: -68

**Comment by YT-Advanced** [comment]
The build fail to boot. And can explain more about your option you choose. This line I can't understand:
> choose stable versions, root/ don't root WSA include Magisk, GAPPS & exclude Amazon App store don't compress the output

Is your option like this ones ?
- Magisk: Stable Version
- GAPPS: Yes
- Magisk: Yes
- Root: No
- Amazon Store: Delete
- Compression: Nope

**Comment by kevstar0** [comment]
Yes my options are like that. (choose to **root** or you may choose **not to root**, I tried both)

**Comment by YT-Advanced** [comment]
@kevstar0 Hmm, WSA fail because of the security level. So I think you should test the lastest version of MagiskOnWSALocal ?
Or if you want to get the build now, you can download this prebuilt build:
https://github.com/YT-Advanced/WSA-Script/releases/latest/

**Comment by kevstar0** [comment]
Been testing newer versions like 2307 intergrated with magisk, GAPPS excluding Amazon and it works as expected, the booting up though could be worrysome especially after signing into google or right after the subsytem finishes installing googleplay libraries after signin. I almost uninstalled it again until I opened WSA files without trying to boot it from an app I installed using ADB. it launched but it struggled. so... i dont know if it is safe to close this issue since the issue is partially remedied

> @kevstar0 Hmm, WSA fail because of the security level. So I think you should test the lastest version of MagiskOnWSALocal ? Or if you want to get the build now, you can download this prebuilt build: https://github.com/YT-Advanced/WSA-Script/releases/latest/
